drought/southern-africa/southern_africa_ipc_map.R

```diff
@@ -0,0 +1,144 @@
+  
+
+sa_country_ipc <- merge(sa_country_shp, sa_ipc_data, by.x = "admin_id", by.y = "admin_fixed", all = TRUE)
+
+### Phase 3+ plot for current projections
+total_plot_fxn <- function(geodata1, geodata2, phase, period, text){
+  
+  # plot
+  phase3 <- geodata1 %>%
+    filter((Phase == phase | is.na(phase)) & (`Validity period` == period | is.na(`Validity period`))) %>%
+    mutate(Number_t = as.numeric(Number)/100000)
+  
+  ggplot() +
+    geom_sf(data = phase3, aes(fill = Number_t)) +
+    geom_sf(data = geodata2, fill = "transparent", linewidth=1) +
+    geom_sf_text(data = geodata2, aes(label = COUNTRY)) +
+    annotate(geom = "text", x = 40, y = -35, label = text, 
+             color = "grey22", size = 4) +
+    # geom_sf_label(aes(label = country_name)) +
+    ggtitle(label = paste0("Acute Food Insecurity IPC Phase ", phase, " (", str_to_title(period), ")"), 
+            subtitle = "Population at Risk in '00,000s") +
+    scale_fill_gradient(low="pink", high="red") +
+    labs(fill = "in '00,000s", x = "", y = "") +
+    theme_hdx()
+  
+}
+
+# 3+ and current period
+total_plot_fxn(geodata1 = sa_country_ipc, geodata2 = sa_shp, 
+         phase = "3+", period = "current", 
+         text = "Zambia - Aug 2023 to Sep 2023\nMadagascar - May 2023 to Sep 2023\nMozambique - Jul 2023 to Sep 2023")
+
+# 4+ and current period
+total_plot_fxn(geodata1 = sa_country_ipc, geodata2 = sa_shp, 
+         phase = "4", period = "current", 
+         text = "Zambia - Aug 2023 to Sep 2023\nMadagascar - May 2023 to Sep 2023\nMozambique - Jul 2023 to Sep 2023")
+
+# 3+ and projected period
+total_plot_fxn(geodata1 = sa_country_ipc, geodata2 = sa_shp, 
+         phase = "3+", period = "projected", 
+         text = "Zambia - Oct 2023 to Mar 2024\nMadagascar - Oct 2023 to Dec 2023\nMozambique - Oct 2023 to Mar 2024")
+
+# 4+ and projected period
+total_plot_fxn(geodata1 = sa_country_ipc, geodata2 = sa_shp, 
+         phase = "4", period = "projected", 
+         text = "Zambia - Oct 2023 to Mar 2024\nMadagascar - Oct 2023 to Dec 2023\nMozambique - Oct 2023 to Mar 2024")
+
+### Percent of Population
+### Phase 3+ plot for current projections
+perc_plot_fxn <- function(geodata1, geodata2, phase, period, text){
+  
+  # plot
+  phase3 <- geodata1 %>%
+    filter((Phase == phase | is.na(phase)) & (`Validity period` == period | is.na(`Validity period`))) 
+  
+  ggplot() +
+    geom_sf(data = phase3, aes(fill = percent_pop)) +
+    geom_sf(data = geodata2, fill = "transparent", linewidth=1) +
+    geom_sf_text(data = geodata2, aes(label = COUNTRY)) +
+    annotate(geom = "text", x = 40, y = -35, label = text, 
+             color = "grey22", size = 4) +
+    # geom_sf_label(aes(label = country_name)) +
+    ggtitle(label = paste0("Acute Food Insecurity IPC Phase ", phase, " (", str_to_title(period), ")"), 
+            subtitle = "Percent of Population at Risk in '00,000s") +
+    scale_fill_gradient(low="pink", high="red") +
```

Comment:
could could consider `label= scales::label_percent()`, also could potentially flip color ramp labels 90 degree  so they are easier to read on a couple?